src/main/java/seedu/address/logic/parser/TypeParsingUtil.java

```diff
@@ -0,0 +1,305 @@
+        String hourMinPattern = "(\\d{1,2}):(\\d{2})";
+        Pattern p = Pattern.compile(hourMinPattern);
+        Matcher m = p.matcher(input);
+        if (m.matches()) {
+            int hour = TypeParsingUtil.parseNum(m.group(1));
+            int min = TypeParsingUtil.parseNum(m.group(2));
+            if (hour > 23 || min > 59) {
+                throw new InvalidInputException(input + " is not a valid time");
+            }
+            return LocalTime.of(hour, min);
+        } else {
+            throw new InvalidInputException(input + " is not a valid time");
+        }
+    }
+    /**
+     * overloading parseTime to take in flagName and parse the flag from the input string
+     */
+    public static LocalTime parseTime(String flagName, String input) throws ParseException {
+        return parseTime(parseFlag(flagName, input));
+    }
+    /**
+     * overloading to not throw exception if the flag is not found when isOptional is true
+     */
+    public static LocalTime parseTime(String flagName, String input, boolean isOptional) throws ParseException {
+        if (isOptional) {
+            try {
+                parseFlag(flagName, input);
+            } catch (FlagNotFoundException e) {
+                return null;
+            }
+        }
+        return parseTime(parseFlag(flagName, input));
+    }
+    private static Integer findMaxDay(int year, int month) {
+        if (month == 2) {
+            if (year % 4 == 0) {
+                return 29;
+            } else {
+                return 28;
+            }
+        } else if (month == 4 || month == 6 || month == 9 || month == 11) {
+            return 30;
+        } else {
+            return 31;
+        }
+    }
+    /**
+     * Parses the date from the input string, which can be in the following formats:
+     * 1. dd, then the date will be the current month and year
+     * 2. mm/dd ,then the date will be the current year
+     * 3. yyyy/mm/dd ,then the date will be the given year
+     * 4. yy/mm/dd ,then the date will be the given year + 2000
+     * @param input the input string where the date is to be parsed from
+     * @return the date parsed
+     * @throws ParseException if the input is not a valid date
+     */
+    public static LocalDate parseDate(String input) throws ParseException {
+        String dayPattern = "(\\d{1,2})";
+        String monthDayPattern = "(\\d{1,2})/(\\d{1,2})";
+        String yearMonthDayPattern = "(\\d{2,4})/(\\d{1,2})/(\\d{1,2})";
+        Pattern dayP = Pattern.compile(dayPattern);
+        Pattern monthDayP = Pattern.compile(monthDayPattern);
+        Pattern yearMonthDayP = Pattern.compile(yearMonthDayPattern);
+        Matcher dayM = dayP.matcher(input);
+        Matcher monthDayM = monthDayP.matcher(input);
+        Matcher yearMonthDayM = yearMonthDayP.matcher(input);
+        if (yearMonthDayM.matches()) {
+            int year = TypeParsingUtil.parseNum(yearMonthDayM.group(1), 0, 9999);
```

Comment:
Would it be better for max to be 99 instead then if adding 2000 in method, in case user has a typo (eg. 233)?

Reply:
Oh, the regular pattern "\d{2,4}" will only match 2 or 4 consecutive numbers, so it would not match in that case

Reply:
Oh I see, thanks for the clarification